llvm/lib/ProfileData/IndexedMemProfData.cpp

```diff
@@ -225,6 +228,8 @@ static Error writeMemProfRadixTreeBased(
   OS.write(0ULL); // Reserve space for the memprof call stack payload offset.
   OS.write(0ULL); // Reserve space for the memprof record payload offset.
   OS.write(0ULL); // Reserve space for the memprof record table offset.
+  if (Version == memprof::Version4)
```

Comment:
Should this check be >= ? Ditto for similar check later. Also, does the version need to be bumped for this change?